h/static/scripts/tests/controllers/autosuggest-dropdown-controller-test.js

```diff
@@ -0,0 +1,413 @@
+
+        let itemContents = `<span class="a-title"> ${listItem.title} </span>`;
+
+        if(listItem.explanation){
+          itemContents += `<span class="an-explanation"> ${listItem.explanation} </span>`;
+        }
+
+        return itemContents;
+      },
+
+      listReducer: function(list, currentInput){
+
+
+        currentInput = (currentInput || '').trim();
+
+        return list.filter((item)=>{
+
+          if(!currentInput){
+            return item;
+          }
+          return item.title.toLowerCase().indexOf(currentInput) >= 0;
+        });
+      },
+
+      onSelect: function(){},
+    };
+
+    const isSuggestionContainerVisible = ()=>{
+      const suggestionContainer = container.querySelector('.' + defaultConfig.classNames.container);
+      return suggestionContainer.classList.contains('is-open');
+    };
+
+    const getListItems = ()=>{
+      const suggestionContainer = container.querySelector('.' + defaultConfig.classNames.container);
+      return suggestionContainer.querySelectorAll('.' + defaultConfig.classNames.item);
+    };
+
+    beforeEach(function () {
+      container = document.createElement('div');
+      container.innerHTML = '<form><input id="input-el"/></form>';
+      document.body.appendChild(container);
+      input = document.getElementById('input-el');
+      form = container.querySelector('form');
+      form.onsubmit = sinon.spy();
+    });
+
+    afterEach(function () {
+      document.body.removeChild(container);
+
+      // clear up spies
+      if('restore' in defaultConfig.listReducer){
+        defaultConfig.listReducer.restore();
+      }
+
+      if('restore' in defaultConfig.onSelect){
+        defaultConfig.onSelect.restore();
+      }
+    });
+
+
+    it('should initialize the controller with correct dom', function(){
+
+      assert.isTrue(form.childNodes.length === 1, 'baseline');
+
+      new AutosuggestDropdownController(input, defaultConfig);
+
+      assert.isFalse(form.childNodes.length === 1, 'initializing should add container to dom');
+
+      const suggestionContainer = container.querySelector('.' + defaultConfig.classNames.container);
+
+      assert.isOk(suggestionContainer);
+      assert.isOk(suggestionContainer.querySelector('.' + defaultConfig.classNames.header));
+      assert.isOk(suggestionContainer.querySelector('.' + defaultConfig.classNames.list));
+      assert.lengthOf(getListItems(), 4);
+
+      assert.lengthOf(suggestionContainer.querySelectorAll('.a-title'), 4, 'reflects our rendering');
+      assert.lengthOf(suggestionContainer.querySelectorAll('.an-explanation'), 4, 'reflects our rendering');
+
+    });
+
+
+    it('opens and closes based on focus status', function (done) {
+
+      new AutosuggestDropdownController(input, defaultConfig);
+
+      assert.isFalse(isSuggestionContainerVisible(), 'basline hidden');
+
+      syn
+        .click(input, () => {
+
+          assert.isTrue(isSuggestionContainerVisible(), 'focus show');
```

Comment:
The last argument to `assert` should be a short message that explains what the error was. eg. "Suggestions not visible when input focused") - something that makes it easy to understand what was wrong if the test fails.